x-pack/test/saved_object_api_integration/common/suites/import.ts

```diff
@@ -8,49 +8,85 @@ import expect from '@kbn/expect';
 import { SuperTest } from 'supertest';
 import { SAVED_OBJECT_TEST_CASES as CASES } from '../lib/saved_object_test_cases';
 import { SPACES } from '../lib/spaces';
-import {
-  createRequest,
-  expectResponses,
-  getUrlPrefix,
-  getTestTitle,
-} from '../lib/saved_object_test_utils';
+import { expectResponses, getUrlPrefix, getTestTitle } from '../lib/saved_object_test_utils';
 import { ExpectResponseBody, TestCase, TestDefinition, TestSuite } from '../lib/types';
 
 export interface ImportTestDefinition extends TestDefinition {
-  request: Array<{ type: string; id: string }>;
+  request: Array<{ type: string; id: string; originId?: string }>;
+  overwrite: boolean;
+  trueCopy: boolean;
 }
 export type ImportTestSuite = TestSuite<ImportTestDefinition>;
 export interface ImportTestCase extends TestCase {
+  originId?: string;
+  expectedNewId?: string;
+  successParam?: string;
   failure?: 400 | 409; // only used for permitted response case
+  fail409Param?: string;
```

Comment:
**question** what are your thoughts about adopting a similar pattern to what I'm attempting in #67644, where `failure` becomes an object with a status code and failure reason? IMO it's more future-proof for when we have to introuce test cases for failure scenarios other than `409`'s:

https://github.com/legrego/kibana/blob/793dc760c1c1951223eac026f9be9e9226e09aeb/x-pack/test/saved_object_api_integration/common/suites/find.ts#L36-L43

Reply:
I agree that would be better. I started working on modifying this but it's not quite as simple for bulk operations like `import`, `bulk_create`, etc...
For these operations, the top-level status code is only ever 200 or 403. So we'll need to do some not-insignificant refactoring for it. And I'd prefer to use the helper methods in `saved_object_test_utils`, but that means we need to change all the test suites at the same time.

If it's OK with you, I'll add this as a task in Phase 2.

Reply:
I'm fine addressing this as part of Phase 2